### PR TITLE
RDoc for module Process

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Adds a "What's Here" section.  This is partly for me to begin to understand this module.  As I add sections in the main discussion, this may become redundant and unnecessary. Therefore I'm marking this as a draft, but also asking @peterzhu2118 for review.

### CODE REVIEW

File: process.c [hunk 1]
```diff
@@ -8570,19 +8570,88 @@ proc_warmup(VALUE _)
...
+ * - ::exit: Initiates process termination by raising exception SystemExit
+ *   (which may be caught).
+ * - ::exit!: Immediately exits the process.
+ * - ::warmup: Notifies the Ruby virtual machine that the boot sequence
+ *   for the application is completed,
+ *   and that the VM may begin optimizing the application.
+ *
+ * === Subprocesses
+ *
+ * - ::_fork: An internal method for forking; should not be called directly.
```
Comment (peterzhu2118) [inline process.c:8617]: We shouldn't document this. It's not a recommended API to use.

**Review by peterzhu2118 — COMMENTED** [review]
I think this document looks good so far.

**Comment by BurdetteLamar** [comment]
Should we say "child process" instead of "subprocess"; seems in keeping with the rest of the world.

File: process.c (hunk 1 above)
Comment (BurdetteLamar) [reply process.c:8617]: My thought, too.  But existing doc does say it can be overridden.

**Comment by peterzhu2118** [comment]
I like "child process" more than "subprocess" to be consistent with the Linux manual.

File: process.c (hunk 1 above)
Comment (BurdetteLamar) [reply process.c]: Removed (from What's Here).

**Comment by BurdetteLamar** [comment]
> I like "child process" more than "subprocess" to be consistent with the Linux manual.

Changed.

File: process.c [hunk 2]
```diff
@@ -8570,19 +8570,87 @@ proc_warmup(VALUE _)
...
- *   * Low-level introspection of other processes, like Process.getpgid, Process.getpriority;
- *   * Management of the current process: Process.abort, Process.exit, Process.daemon, etc.
- *     (for convenience, most of those are also available as global functions
- *     and module functions of Kernel);
- *   * Creation and management of child processes: Process.fork, Process.spawn, and
- *     related methods;
- *   * Management of low-level system clock: Process.times and Process.clock_gettime,
- *     which could be important for proper benchmarking and other elapsed
- *     time measurement tasks.
+ * \Module \Process represents a process in the underlying operating system.
```
Comment (nobu) [inline process.c:8573]: To escape `Module` is that it is a common word, but `Process` is not a common word here.
I think it is better to mark up as code.
```suggestion
 * \Module +Process+ represents a process in the underlying operating system.
```

**Comment by nobu** [comment]
Frankly speaking, I'm not sure why you want to avoid to link to classes/modules.

File: process.c (hunk 2 above)
Comment (BurdetteLamar) [reply process.c:8573]: @nobu, we have hundreds (if not thousands) of instances like \Process.  If ok with you, I'd like to defer this and propose a usage in the documentation guide.

File: process.c [hunk 3]
```diff
@@ -8570,19 +8570,87 @@ proc_warmup(VALUE _)
...
+ * - ::exit!: Immediately exits the process.
+ * - ::warmup: Notifies the Ruby virtual machine that the boot sequence
+ *   for the application is completed,
+ *   and that the VM may begin optimizing the application.
+ *
+ * === Child Processes
+ *
+ * - ::detach: Guards against a child process becoming a zombie.
+ * - ::fork: Creates a child process.
+ * - ::kill: Sends a given signal to processes.
```
Comment (nobu) [inline process.c:8619]: `kill` is not only for child processes.
But I can't say it should not be here since it is the typical usage...

**Comment by BurdetteLamar** [comment]
@nobu @peterzhu2118 I think all is resolved;  ok to squash and merge?

File: process.c [hunk 4]
```diff
@@ -8570,19 +8570,87 @@ proc_warmup(VALUE _)
...
+ * - ::spawn: Creates a chile process.
```
Comment (peterzhu2118) [inline process.c:8620]: Typo

```suggestion
 * - ::spawn: Creates a child process.
```

**Review by peterzhu2118 — APPROVED** [review]

**Comment by BurdetteLamar** [comment]
> Frankly speaking, I'm not sure why you want to avoid to link to classes/modules.

@nobu, are you content with https://docs.ruby-lang.org/en/master/contributing/documentation_guide_md.html#label-Auto-Linking?  If so, I'll be more strict with myself.

**Pull request merged**